### ISSUE TITLE
No events are associated with crashes

### ISSUE THREAD

**Issue by ufna** [Microsoft/appcenter#311]
**What App Center service does this affect?**
Crashes + Analytics

**Describe the bug**
I've tested events before crash, so I've sent few events and then generated test crash (NDK one). Unfortunately, at report page I got "It looks like the App Center Analytics SDK was not installed or enabled at the time of this crash." and no events.

**To Reproduce**
1. Launch app
2. Send few Events
3. Crash app
4. Relaunch and wait 30-40 sec
5. Check the crash once it will be processed

**Expected behavior**
Events are listed on the Events tab in crash details.

**Screenshots**

**Smartphone (please complete the following information):**
 - Device: Galaxy S8
 - OS: Android 9
 - Version: 1.11.4

**Additional context**
Device logs are provided to support team via "Conversation" ui.

**Comment by J-Swift** [comment]
I've had the same issue on both iOS/Android for a little while and just hadn't gotten around to reporting it. I had events associated with my crashes when I initially set them up, but at some point they stopped being sent with them.

**Comment by jashan07** [comment]
Same here, the events I register appear in the events tab under 'Analytics' so they are definitely being sent out to the appcenter. For some reason they are mostly missing in my crashreport.

**Comment by vrwim** [comment]
Adding a "me-too" to hope this gets some attention.

The events appear in the log flow (as does the unhandled error), but they don't appear in the "events" tab of a report.

I am using Xamarin.iOS, AppCenter 1.12.0 (via NuGet), iPhone X on iOS 12.4 Beta

**Comment by mylemans** [comment]
Same issue here with macOS. Support directed me to this issue. Events are logged properly, and on one occasion I kept an eye on the event log flow and I saw the events coming in, followed by the 'crash' report, but even for that one it cannot show the event log for that crash.

It said 'collecting' for a long time, and now is telling me I must not have had the analytics SDK enabled, while the proof is there I had it on.

**Comment by mylemans** [comment]
The older ones do not, but one of my test crashed today does have the events attached to the crash. Was something fixed on the backend?

**Comment by mylemans** [comment]
And it is broken again.. Come on guys!

"It looks like the App Center Analytics SDK was not installed or enabled at the time of this crash."

It is clearly visible in the event flow & tracked events in general...

**Comment by vrwim** [comment]
It works for some users and not for others I have noticed, same app version, same crash. Some show the error message that the SDK is not installed, others show the events correctly.

**Comment by mylemans** [comment]
Now they appear, maybe increase the delay before showing the analytics SDK message if it is a processing delay?

**Comment by AlexNajem** [comment]
This is pretty important. Can’t believe it’s getting ignored this long.

**Comment by ihorkaralash** [comment]
I have exactly the same issue, please fix it as soon as possible

**Comment by freddyhm** [comment]
Same issue as @vrwim "It works for some users and not for others I have noticed, same app version, same crash."

**Comment by taspeotis** [comment]
Same here, we have the events in Analytics but they're not in Crashes.

**Comment by JakubOleksy** [comment]
@matt-gibbs @yoannchaudet FYI

@freddyhm @taspeotis would it be possible for you to contact support directly in the product? This would help us make sure we get you the right help quickly.

**Comment by WhitePhoera** [comment]
support sent me here, any news on that?

**Comment by felipeogarcia** [comment]
+1

**Comment by gonz01d** [comment]
@JakubOleksy @patniko I am also experiencing this issue and have reported it to Support directly as was suggested - their answer was to refer me back to this issue again 🤷🏻‍♂️ which some 7 months after initially being reported seems to a perfect circle of inaction - are you able to confirm the issue is being investigated?  

I'm happy to volunteer as a use case to try and help resolve the problem If it helps

**Comment by jaromirBergiel** [comment]
+1 | Breadcrumbs (Events before a crash) still doesn't work.

**Comment by yoannchaudet** [comment]
Breadcrumbs has been unreliable in the past because of scaling issues. Some work already happened to improve the situation and more work will come.

Breadcrumbs should be processing fine (as of the last 24 hours). Also note that the association events to crashes only happens if the beginning of the session and the crashe's ingestion by our backend are not separated by more than 48 hours (two days look back).

**Comment by taspeotis** [comment]
> Also note that the association events to crashes only happens if the beginning of the session and the crashe's ingestion by our backend are not separated by more than 48 hours (two days look back).

So what you're saying is if the app crashes, but users don't re-open it within 48h for the crash to be reported, breadcrumbs will not happen? Just for my own understanding, thanks.

**Comment by mylemans** [comment]
Our users tend to keep our app open for days on macOS, in the background. If that is 48h from startup till a crash being uploaded, then we could be missing a lot of info. Is this the case?

This would mean that AppCenter events are pointless for long running apps.

**Comment by tgriggs-4tell** [comment]
I am also seeing this issue, but on a limited basis. We had one user experience three legitimate crashes (not test crashes), on iOS. The first crash came with events, and the next two did not (they had the message saying that App Center was not installed or running). The three crashes all happened within a one hour timespan, so it definitely was not outside of the 48 hour window.

**Comment by mylemans** [comment]
Again 3 unique crash types, 6 reports and none of them have events logged... 

"It looks like the App Center Analytics SDK was not installed or enabled at the time of this crash." - it certainly is installed..

Some feedback or progress would be appreciated.

**Comment by ericc0504** [comment]
This issue happens to me too. Hope that it will be resolved soon.

**Comment by natevarghese** [comment]
+1

**Comment by hallamw** [comment]
make that +2

**Comment by JakubOleksy** [comment]
@yoannchaudet @scottdensmore are there any known issues?

**Comment by cassarreb** [comment]
+1

**Comment by BosseGr** [comment]
+1

**Comment by yoannchaudet** [comment]
> So what you're saying is if the app crashes, but users don't re-open it within 48h for the crash to be reported, breadcrumbs will not happen? Just for my own understanding, thanks.

@taspeotis This is correct. When we receive a crash, we associate it with its trail of events if those (along with the beginning of the session) were ingested by our backend within 48 hours.

Sorry I missed the activity on this thread. By the look of our internal telemetry we are still experiencing issues (and missing alerting)... We do process a ton of breadcrumbs (how we call the chain of events preceding a crash) but we also skip many as we have capacity issue at the moment. Bringing this issue to stakeholders.

**Comment by hibob224** [comment]
+1 Tested several crashes but always seem to get the following message after a while on the event tab

> It looks like the App Center Analytics SDK was not installed or enabled at the time of this crash.

**Comment by TomSoderling** [comment]
Me too. I'm using version 2.5.0 the AppCenter.Analytics NuGet package

**Comment by alex-vit** [comment]
This is one of the key selling points for me. +1.

**Comment by yoannchaudet** [comment]
Update on this.

We mitigated on 12/17 on our side so we are now processing breadcrumbs again. We do have the 48 hours limit I mentioned [above](https://github.com/microsoft/appcenter/issues/311#issuecomment-564154432). But given we ingest more than 90% of all crashes within 24 hours, this limit does not discard many breadcrumbs at the moment. If you are not getting any breadcrumbs for your app and think something is wrong, please do reach out to [our support](https://intercom.help/appcenter/en/articles/1698824-getting-help-with-app-center).

We have some more work planned around breadcrumbs. Short term we will be increasing our backend capacity so we can process more breadcrumbs and ideally increase our look back setting further.

Longer term, we are thinking about implementing breadcrumbs (the trail of events preceding a crash) right in the SDK. This would eliminate this 48 hours technical limit. AND we could finally expose breadcrumbs for handled errors.

**Comment by kenneth-kano** [comment]
This is broken again.

**Comment by MichaelLHerman** [comment]
Broken for me as well.  Events showing up on iOS crashes, but not on Android crashe.
>"It looks like the App Center Analytics SDK was not installed or enabled at the time of this crash.
If you haven’t already, [install the Analytics SDK](https://docs.microsoft.com/en-us/mobile-center/analytics) to help better understand and replicate future crashes."

Yet I have events under Analytics | Events
